htdocs/xoops_lib/modules/protector/class/gtickets.php

```diff
@@ -119,7 +119,11 @@ public function issue($salt = '', $timeout = 1800, $area = '')
             global $xoopsModule;
 
             if ('' === $salt) {
-                $salt = '$2y$07$' . str_replace('+', '.', base64_encode(mcrypt_create_iv(16, MCRYPT_DEV_URANDOM)));
+				if (version_compare(PHP_VERSION, '7.0.0') >= 0) {
+					$salt = '$2y$07$' . str_replace('+', '.', base64_encode(random_bytes(16)));
```

Comment:
https://github.com/paragonie/random_compat  is a required library for some time. It provides a polyfill  for random_bytes() if needed. 

This line is the only one needed as it will work for for all supported PHP versions. Eliminate the version_compare to keep the code cleaner.

Reply:
Thank you for this information, I did not know that this library was used in xoops